qp_klp/process_amplicon_job.py

```diff
@@ -262,24 +264,62 @@ def process_amplicon(mapping_file_path, qclient, run_identifier, out_dir,
         job_id,
         is_amplicon=True)
 
+    touched_studies_prep_info = defaultdict(list)
+
     if not skip_exec:
         gpf_job.run(callback=status_line.update_job_step)
-        # if seq_pro is run, prep_file_paths will be populated by run().
-        prep_file_paths = gpf_job.prep_file_paths
+
         # concatenate the lists of paths across all study_ids into a single
-        # list.
-        pfp_list = list(chain.from_iterable(prep_file_paths.values()))
-    else:
-        # under testing conditions, gpf_job.prep_file_paths will not be
-        # populated. Generate pfp_list from walking the expected location.
-        pfp_list = []
-        for root, dirs, files in walk(join(pipeline.output_path,
-                                           'GenPrepFileJob', 'PrepFiles')):
-            for prep_file in files:
-                if prep_file.endswith('.tsv'):
-                    pfp_list.append(join(root, prep_file))
+        # list. Replace sample-names w/tube-ids in all relevant prep-files.
+        preps = list(chain.from_iterable(gpf_job.prep_file_paths.values()))
+        map_sample_names_to_tube_ids(preps, sn_tid_map_by_project)
+
+        # if seq_pro is run, prep_file_paths will be populated by run().
+        for study_id in gpf_job.prep_file_paths:
+            for prep_file_path in gpf_job.prep_file_paths[study_id]:
+                metadata_dict = pd.read_csv(prep_file_path,
+                                            delimiter='\t').to_dict('index')
 
-    map_sample_names_to_tube_ids(pfp_list, sn_tid_map_by_project)
+                # determine data_type based on target_gene column.
+                for key in {'16S', '18S', 'ITS'}:
+                    if key in metadata_dict[0]['target_gene']:
```

Comment:
This should be: `if key in metadata_dict[list(metadata_dict.keys())[0]]['target_gene']:`

Reply:
I don't follow your question but just to clear the issue is `metadata_dict` is a dict and not a dataframe so we need to get the first key of the dict, then get the 'target_gene' of that key.

Reply:
Ah I see what you're saying, ty!